### PR TITLE
[AC-1416] Expose Organization Fingerprint

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Type of change
- [ ] Bug fix
- [X] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Code changes
- **account.component.html**: Removed bootstrap classes, Updated to reactive forms, added account fingerprint component, updated to component library components/fields
- **account.component.ts**: Updated to reactive forms, updated params subscription to use desired observable pattern, updated `submit` to use handler pattern for CL async operations.
- **organization-settings.module**: Added `AccountFingerprintComponent` to the import list
- **profile.component.html**: replaced boilerplate with new component selector
- **profile.component.ts**: Updated class to store `fingerprintMaterial ` in variable for template access
- **account-fingerprint.component.html/ts**: Created standalone component to minimize html boilerplate for showing fingerprint phase in the profile/org account settings
- **loose-components.module**: Added `AccountFingerprintComponent` to the import list for usage in the `profile-component` 
- **messages.json**: Added new fingerprint phrase label
- **crypto.service**: Updated parameter to better reflect the desired input (fingerprintMaterial)
- **tailwind.config.base.js**: Added theme extensions to emulate `col-6` bootstrap functionality

## Screenshots
Screenshot 2023-06-02 at 4 10 49 PM

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/components/tailwind.config.base.js [hunk 1]
```diff
@@ -92,7 +92,11 @@ module.exports = {
       maxWidth: ({ theme }) => ({
         ...theme("width"),
         "90vw": "90vw",
+        "1/2": "50%", // emulates col-6
```
Comment (Hinton) [inline libs/components/tailwind.config.base.js:95]: You should already have this through the `width`, https://tailwindcss.com/docs/width

File: libs/components/tailwind.config.base.js [hunk 2]
```diff
@@ -92,7 +92,11 @@ module.exports = {
...
       }),
+      flex: {
+        6: "0 0 50%", // emulates col-6
+      },
```
Comment (Hinton) [inline libs/components/tailwind.config.base.js:99]: You probably want to use https://tailwindcss.com/docs/grid-template instead?

**Review by Hinton — COMMENTED** [review]
Feel free to ping me to discuss best approaches for making grids in tailwind.

File: libs/components/tailwind.config.base.js (hunk 1 above)
Comment (vincentsalucci) [reply libs/components/tailwind.config.base.js]: Great call - I was hyper-focused on the `max-width` property. Removed.

File: libs/components/tailwind.config.base.js (hunk 2 above)
Comment (vincentsalucci) [reply libs/components/tailwind.config.base.js]: Using `grid` did accomplish the job here. Removed.

File: apps/web/src/app/admin-console/organizations/settings/account.component.html [hunk 3]
```diff
@@ -70,23 +55,23 @@ <h1>{{ "apiKey" | i18n }}</h1>
...
 </ng-container>
-<div class="secondary-header text-danger border-0 mb-0">
-  <h1>{{ "dangerZone" | i18n }}</h1>
+<div class="tw-mt-16 tw-pb-2.5">
+  <h1 class="!tw-text-danger">{{ "dangerZone" | i18n }}</h1>
 </div>
-<div class="card border-danger">
-  <div class="card-body">
+<div class="tw-rounded tw-border tw-border-solid tw-border-danger-500 tw-bg-background">
+  <div class="tw-p-5">
```
Comment (Hinton) [inline apps/web/src/app/admin-console/organizations/settings/account.component.html:69]: You can eliminate the inner div, by moving the padding to the outer div.

File: apps/web/src/app/admin-console/organizations/settings/account.component.html [hunk 4]
```diff
@@ -9,59 +9,44 @@ <h1>{{ "organizationInfo" | i18n }}</h1>
...
-          class="form-control"
-          type="text"
-          name="Name"
-          [(ngModel)]="org.name"
-          [disabled]="selfHosted"
-        />
+<form *ngIf="org && !loading" #form [bitSubmit]="submit" [formGroup]="formGroup">
+  <div class="-tw-mx-2.5 tw-grid tw-grid-cols-2 tw-gap-5">
+    <div class="tw-pl-2.5">
+      <div class="tw-mb-4">
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.html]: Removed. 👍

File: apps/web/src/app/admin-console/organizations/settings/account.component.html [hunk 5]
```diff
@@ -9,59 +9,44 @@ <h1>{{ "organizationInfo" | i18n }}</h1>
...
-        <input
-          id="name"
-          class="form-control"
-          type="text"
-          name="Name"
-          [(ngModel)]="org.name"
-          [disabled]="selfHosted"
-        />
+<form *ngIf="org && !loading" #form [bitSubmit]="submit" [formGroup]="formGroup">
+  <div class="-tw-mx-2.5 tw-grid tw-grid-cols-2 tw-gap-5">
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.html]: Removed. 👍

File: apps/web/src/app/admin-console/organizations/settings/account.component.html (hunk 3 above)
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.html]: Removed. 👍

**Comment by vincentsalucci** [comment]
@bitwarden/team-platform-dev : updated the `crypto.service -> getFingerprint` method parameter to `fingerprintMaterial` to better represent the potential input options.

**Review by Hinton — APPROVED** [review]

File: apps/web/src/app/admin-console/organizations/settings/account.component.html [hunk 6]
```diff
@@ -9,87 +7,60 @@ <h1>{{ "organizationInfo" | i18n }}</h1>
   ></i>
   <span class="sr-only">{{ "loading" | i18n }}</span>
```
Comment (shane-melton) [inline apps/web/src/app/admin-console/organizations/settings/account.component.html:8]: nit: I think we're also switching these over to `tw-sr-only`

File: apps/web/src/app/admin-console/organizations/settings/account.component.html [hunk 7]
```diff
@@ -9,87 +7,60 @@ <h1>{{ "organizationInfo" | i18n }}</h1>
...
-  </div>
+  <h1 bitTypography="h1" class="tw-mt-16 tw-pb-2.5">{{ "apiKey" | i18n }}</h1>
   <p>
     {{ "apiKeyDesc" | i18n }}
     <a href="https://docs.bitwarden.com" target="_blank" rel="noopener">
       {{ "learnMore" | i18n }}
     </a>
   </p>
-  <button type="button" class="btn btn-outline-secondary" (click)="viewApiKey()">
+  <button type="button" bitButton (click)="viewApiKey()">
```
Comment (shane-melton) [inline apps/web/src/app/admin-console/organizations/settings/account.component.html:48]: question: Should we also specify the button type for these two buttons?

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts [hunk 8]
```diff
@@ -52,34 +75,72 @@ export class AccountComponent {
...
+      .pipe(
+        map((params) => this.organizationService.get(params.organizationId)),
+        switchMap(async (organization) => {
+          // Set domain level organization variables
+          this.organizationId = organization.id;
+          this.canEditSubscription = organization.canEditSubscription;
+          this.canUseApi = organization.useApi;
+
+          try {
+            // Retrieve OrganizationResponse for form population
```
Comment (shane-melton) [inline apps/web/src/app/admin-console/organizations/settings/account.component.ts:95]: suggestion: We could parallelize these two back to back API calls by returning a `combineLatest` of the two and then updating the `formGroup` value, controls, and loading flag in the `subscribe` body.

Slightly similar example of using `combineLatest`: https://github.com/bitwarden/clients/blob/cd85a4c03903c9e9aa92106d77a847e988a4fd2e/apps/web/src/app/admin-console/organizations/manage/groups.component.ts#L139-L174

**Review by shane-melton — COMMENTED** [review]
The fingerprint component looks really good to me! I only had a few suggestions regarding the account component updates.

File: apps/web/src/app/admin-console/organizations/settings/account.component.ts [hunk 9]
```diff
@@ -52,34 +75,72 @@ export class AccountComponent {
...
+    this.destroy$.complete();
   }
 
-  async submit() {
+  submit = async () => {
+    this.formGroup.markAllAsTouched();
+    if (this.formGroup.invalid) {
+      return;
+    }
+
```
Comment (shane-melton) [inline apps/web/src/app/admin-console/organizations/settings/account.component.ts:138]: suggestion: We can remove the `try...catch` as that's already handled by the `bitSubmit` directive and has the benefit of showing an error toast via the `validationService`.

https://github.com/bitwarden/clients/blob/cd85a4c03903c9e9aa92106d77a847e988a4fd2e/libs/components/src/async-actions/bit-submit.directive.ts#L44-L50

Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.ts:138]: Ah - great catch. Thank you!

File: apps/web/src/app/admin-console/organizations/settings/account.component.html (hunk 6 above)
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.html:8]: I missed this: thank you!

Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/settings/account.component.html]: Updated!

**Review by shane-melton — DISMISSED** [review]
Looks great! That observable change looks very clean now!

**Comment by vincentsalucci** [comment]
QA approved - removing label and merging!

**Review by shane-melton — APPROVED** [review]

**Pull request merged**